menu/detect_mfa.go

```diff
@@ -0,0 +1,171 @@
+package menu
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/fatih/color"
+)
+
+var (
+	ErrInvalidCredentials = errors.New("Credentials are not valid")
+	ErrNotAUser           = errors.New("Credentials are not for a user principal")
+)
+
+type DetectMFAMenu struct {
+	*Menu
+}
+
+func (m *DetectMFAMenu) Handler() error {
+	fmt.Println()
+	fmt.Printf("Checking for MFA devices...\n")
+	mfaDevices, err := m.getMFADevices()
+	if err != nil {
+		fmt.Printf("Unable to check for MFA devices.\n")
+		fmt.Println()
+		return err
+	}
+
+	switch len(mfaDevices) {
+	case 0:
+		fmt.Printf("No MFA devices found.\n")
+		fmt.Println()
+
+	case 1:
+		fmt.Printf("1 MFA device found.\n")
+		fmt.Println()
+		return m.promptForSingle(mfaDevices[0])
+
+	default:
+		fmt.Printf("%d MFA devices found.\n", len(mfaDevices))
+		fmt.Println()
+		return m.promptForMultiple(mfaDevices)
+	}
+
+	return nil
+}
+
+func (m *DetectMFAMenu) promptForSingle(mfaDevice *iam.MFADevice) error {
+	for {
+		warningColor.Printf("  %s\n", *mfaDevice.SerialNumber)
+
+		input, err := interaction.ReadPrompt("Would you like to use this MFA device? (Y/n): ")
+		if err != nil {
+			return err
+		}
+
+		switch strings.ToLower(input) {
+		case "", "y", "yes":
+			m.Vault.AWSKey.MFA = *mfaDevice.SerialNumber
+			return nil
+
+		case "n", "no":
+			m.Vault.AWSKey.MFA = ""
+			return nil
+
+		default:
+			fmt.Println()
+			color.Red("Response not recognized. Please enter 'y' or 'n'.")
+			fmt.Println()
+			continue
+		}
+	}
+}
+
+func (m *DetectMFAMenu) promptForMultiple(mfaDevices []*iam.MFADevice) error {
+	for {
+		warningColor.Println("  0) None")
+		for i, mfaDevice := range mfaDevices {
+			warningColor.Printf("  %d) %s\n", i+1, *mfaDevice.SerialNumber)
+		}
+
+		input, err := interaction.ReadPrompt("Select the MFA device to use: ")
+		if err != nil {
+			return err
+		}
+
+		index, err := strconv.ParseInt(input, 10, 64)
+		if err != nil {
+			fmt.Println()
+			color.Red("Response not recognized. Please enter a value from 0 to %d.", len(mfaDevices))
+			fmt.Println()
+			continue
+		}
+
+		if index < 0 || int(index) > len(mfaDevices) {
```

Comment:
we cast index to an int in one check here but not the other.  do we need to do this for both?